### PR TITLE
fix: Fix page.firstSortValues returned by the search API

### LINKED ISSUE #26080: Follow-up: Fix page.firstSortValues returned by the search API
**Describe the bug**

Follow-up of https://github.com/camunda/camunda/pull/26082

Currently in the [search result transformer](https://github.com/camunda/camunda/blob/94e8ae400f328a1fb49ea0882242bd8494f40c53/search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/query/SearchQueryResultTransformer.java#L33-L38), we get only the sortValues of the last item of the page.

We also need to get the sortValues of the first item to pass it into firstSortValues returned in the response, to be used to fetch the previous page

### PR DESCRIPTION
**Pull request by houssain-barouni**
## Description

We need to return the `firstSortValues` in the search response so that it can be used to fetch the previous page

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #26080

### CODE REVIEW

File: search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/query/SearchQueryResultTransformer.java [hunk 1]
```diff
@@ -16,30 +16,31 @@
...
 
   public SearchQueryResultTransformer(final ServiceTransformer<T, R> documentToEntityMapper) {
     this.documentToEntityMapper = documentToEntityMapper;
   }
 
-  @Override
-  public SearchQueryResult<R> apply(final SearchQueryResponse<T> value) {
-    final var hits = value.hits();
+  public SearchQueryResult<R> apply(final SearchQueryResponse<T> value, final boolean reverse) {
+    final var hits = reverse ? value.hits().reversed() : value.hits();
```
Comment (tmetzke) [inline search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/query/SearchQueryResultTransformer.java:27]: ❓ Maybe I overlooked it, but can you explain why this reversing is needed? There is no explanation in the commit, code, PR, or issue description. It is not an obvious change and should be documented if possible.

**Review by tmetzke — APPROVED** [review]
👍 The changes regarding firstSortValues look sound to me, thanks for tackling this right away 🚀 
🔧 The reversing option is not an obvious change and I don't see any explanation for it. Would be helpful to document why we have this and what this is needed for, if possible.

File: search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/query/SearchQueryResultTransformer.java (hunk 1 above)
Comment (houssain-barouni) [reply search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/query/SearchQueryResultTransformer.java:27]: yes sure, it is related to https://camunda.slack.com/archives/C06UKS51QV9/p1734115692563139 . I will add a comment explaining it

Comment (tmetzke) [reply search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/query/SearchQueryResultTransformer.java:27]: Thanks! That should be helpful for anyone coming across this in the future.

Comment (houssain-barouni) [reply search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/query/SearchQueryResultTransformer.java:39]: done 👍

**Review by stoerti — APPROVED** [review]
RDBMS changes look good to me 👍

File: search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/query/SearchQueryResultTransformer.java (hunk 1 above)
Comment (tmetzke) [reply search/search-client-query-transformer/src/main/java/io/camunda/search/clients/transformers/query/SearchQueryResultTransformer.java:39]: Really good explanation, this is DEFINITELY going to help in the future! 🚀

**Pull request merged**